transaction/src/transaction_view/transaction_perspective.rs

```diff
@@ -0,0 +1,34 @@
+use std::collections::BTreeMap;
+
+use penumbra_crypto::{
+    memo::{MemoCiphertext, MemoPlaintext},
+    note, Note, Nullifier, PayloadKey,
+};
+
+/// This represents the data to understand an individual transaction without
+/// disclosing viewing keys.
+pub struct TransactionPerspective {
+    /// List of per-action payload keys. These can be used to decrypt
+    /// the notes, swaps, and memo keys in the transaction.
+    ///
+    /// One-to-one correspondence between:
+    /// * Output and note,
+    /// * Swap and note (NFT),
+    ///
+    /// There is not a one-to-one correspondence between SwapClaim and notes,
+    /// i.e. there are two notes per SwapClaim.
+    ///
+    /// For outputs, we can use the PayloadKey associated with that output
+    /// to decrypt the wrapped_memo_key, which will be used to decrypt the
+    /// memo in the transaction. This needs to be done only once, because
+    /// there is one memo shared between all outputs.
+    pub payload_keys: BTreeMap<note::Commitment, PayloadKey>,
+    /// Mapping of nullifiers spent in this transaction to notes.
+    pub spend_nullifiers: BTreeMap<Nullifier, Note>,
+
+    pub memo_cipher_text: Option<MemoCiphertext>,
+
+    pub memo_plaintext: Option<MemoPlaintext>,
```

Comment:
I think I'm missing something -- why are the memo plaintext and ciphertext included in the `TransactionPerspective`?

First, why do we need to record both the ciphertext and the plaintext -- if we have the ciphertext, can't we decrypt it to obtain the plaintext?

Second, why do we need the ciphertext -- won't that be in the Transaction we're viewing with this Perspective?

Reply:
We don't, this is just a couple placeholder fields to avoid refactoring the IsAction decryption method or the TransactionView further, since the Transaction itself isn't available on the method that actually decrypts the Memo. It could be, but this was a question mark. I have no strong opinions about how it should be handled, but refactoring the Transaction-level decryption method was going to be more work than was justified to hold this up further.